components/layout/text.rs

```diff
@@ -230,6 +231,14 @@ impl TextRunScanner {
                         None => false
                     };
 
+                    if !initialized_run_info {
```

Comment:
During the iteration every time we start to deal with a fragment if the font changed we may create an empty mapping, and when we are done with a fragment we flush all the remaining text to mapping, so instead using a temp variable for the whole iteration we should check if end_position is zero.